From 16cd8310ffbb5a728abb8eb9af6a14645f834713 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:19:05 -0400
Subject: [net] netfilter: nft_nat: don't dump port information if unset

Message-id: <20141021161905.22888.210.stgit@localhost.localdomain>
Patchwork-id: 97971
O-Subject: [RHEL7 net PATCH 074/124] netfilter: nft_nat: don't dump port information if unset
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 915136065b7ca75af4cae06281e4dc43926edbfe

 commit 915136065b7ca75af4cae06281e4dc43926edbfe
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Fri Jun 13 13:45:38 2014 +0200

    netfilter: nft_nat: don't dump port information if unset

    Don't include port information attributes if they are unset.

    Reported-by: Ana Rey <anarey@gmail.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nft_nat.c b/net/netfilter/nft_nat.c
index a0195d2..79ff58c 100644
--- a/net/netfilter/nft_nat.c
+++ b/net/netfilter/nft_nat.c
@@ -175,12 +175,14 @@ static int nft_nat_dump(struct sk_buff *skb, const struct nft_expr *expr)
  if (nla_put_be32(skb,
     NFTA_NAT_REG_ADDR_MAX, htonl(priv->sreg_addr_max)))
   goto nla_put_failure;
- if (nla_put_be32(skb,
-    NFTA_NAT_REG_PROTO_MIN, htonl(priv->sreg_proto_min)))
-  goto nla_put_failure;
- if (nla_put_be32(skb,
-    NFTA_NAT_REG_PROTO_MAX, htonl(priv->sreg_proto_max)))
-  goto nla_put_failure;
+ if (priv->sreg_proto_min) {
+  if (nla_put_be32(skb, NFTA_NAT_REG_PROTO_MIN,
+     htonl(priv->sreg_proto_min)))
+   goto nla_put_failure;
+  if (nla_put_be32(skb, NFTA_NAT_REG_PROTO_MAX,
+     htonl(priv->sreg_proto_max)))
+   goto nla_put_failure;
+ }
  return 0;
 
 nla_put_failure:
-- 
1.7.1